Starforge/Editor/Autotiler.cs

```diff
@@ -0,0 +1,261 @@
+﻿using Microsoft.Xna.Framework;
+using Starforge.Core;
+using Starforge.Editor.Render;
+using Starforge.Editor.UI;
+using Starforge.Map;
+using Starforge.Mod.Content;
+using Starforge.Util;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+
+namespace Starforge.Editor {
+    /// <summary>
+    /// Generates textures based on tiles.
+    /// </summary>
+    public class Autotiler {
+        /// <summary>
+        /// A dictionary of all registered tilesets.
+        /// </summary>
+        private Dictionary<char, Tileset> Tilesets;
+
+        /// <summary>
+        /// A byte array used to generate random tile textures.
+        /// </summary>
+        private byte[] TileRand;
+
+        /// <summary>
+        /// Creates an autotiler.
+        /// </summary>
+        /// <param name="xmlPath">The path to the file containing the tileset definitions.</param>
+        public Autotiler(string xmlPath) {
+            // Create tile randomization array
+            TileRand = new byte[1048576];
```

Comment:
This seems to be pretty overkill, since you only ever use [x+yInc] in this array, which will probably not be this big. So maybe just make this a lot smaller, like 512 or something (also make that a constant please xD) and then when you get a value from it just do [(x + yInc) % size]